src/pykiso/test_setup/config_registry.py

```diff
@@ -18,97 +18,225 @@
 .. currentmodule:: config_registry
 
 """
+from __future__ import annotations
 
-from typing import Any
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple, Type
 
 from .dynamic_loader import DynamicImportLinker
 
+if TYPE_CHECKING:
+    from pykiso.auxiliary import AuxiliaryCommon
+    from pykiso.types import (
+        AuxiliaryAlias,
+        AuxiliaryConfig,
+        ConfigDict,
+        ConnectorAlias,
+        ConnectorConfig,
+    )
+
 
 class ConfigRegistry:
     """Register auxiliaries with connectors to provide systemwide import
     statements.
+
+    Internally patch the user-configuration if multiple auxiliaries share
+    a same communication channel.
     """
 
     _linker = None
 
+    @staticmethod
+    def _make_proxy_channel_config(
+        aux_name: AuxiliaryConfig, multiprocessing: bool
+    ) -> Tuple[ConnectorAlias, ConnectorConfig]:
+        """Craft the configuration dictionary for a proxy communication
+        channel to attach to the auxiliary instead of the 'physical' channel.
+
+        :param aux_name: name of the auxiliary to which the proxy channel
+            should be plugged.
+        :param multiprocessing: whether the auxiliary is multiprocessing-
+            based or thread-based.
+        :return: the resulting proxy channel name and its configuration as
+            as a tuple.
+        """
+        from pykiso.lib.connectors.cc_mp_proxy import CCMpProxy
```

Comment:
Humm I don't really like it why do we this import there

Reply:
Otherwise we get circular dependencies, it's this way or we have to reorder the imports in the root `__init__.py` (which will be a hell to maintain with isort)

Reply:
(`pykiso` imports `auxiliary` which import `config_registry` which would import `proxy_auxiliary` which imports `auxiliary`)